Makefile

```diff
@@ -0,0 +1,384 @@
+RUST_LOG ?= info
+# you can use devnet | testnet | mainnet
+chain ?= devnet
+# set the data path when the claster run
+data_path ?= /tmp/data
+# you can add a prefix such as node0 | node1 to create mulpile clusters on one host
+prefix ?=
+sgx_mode ?= HW
+# rebuild
+rebuild ?=
+ifeq ($(rebuild)x, x)
+	CARGO_BUILD_CMD = cargo build --release
+	MAKE_CMD := make
+else
+	CARGO_BUILD_CMD = cargo clean && cargo build --release
+	MAKE_CMD = make clean && make
+endif
+
+base_port ?= 26650
+TX_QUERY_PORT       = $(shell expr $(base_port) + 1)
+TENDERMINT_P2P_PORT = $(shell expr $(base_port) + 6)
+TENDERMINT_RPC_PORT = $(shell expr $(base_port) + 7)
+CLIENT_RPC_PORT     = $(shell expr $(base_port) + 9)
+
+# the chain version, such as v0.1.0, v0.2.0.
+tag ?=
+# if the tag not set, it will use current tag, if current code is not checkouted to a tag, it will use `develop`
+ifeq ($(tag)x, x)
+	TAG = $(shell git describe --exact-match --tags $(git log -n1 --pretty='%h') 2>/dev/null || echo develop)
+else
+	TAG = $(tag)
+endif
+
+# docker's network
+NETWORK = $(prefix)crypto-chain
+
+# if the TAG like v0.1.0, v0.2.0, we download the binary file from github
+ifeq ($(TAG), develop)
+	DOWNLAD_URL=
+else
+	DOWNLOAD_URL=$(shell curl -s https://api.github.com/repos/crypto-com/chain/releases \
+            | grep browser_download_url \
+            | grep download/$(tag)/ \
+            | cut -d '"' -f 4 || echo "")
+endif
+
+APP_HASH := $(shell cat docker/config/$(chain)/tendermint/genesis.json | python -c "import json,sys;obj=json.load(sys.stdin);print(obj['app_hash'])")
+
+ifeq ($(chain), devnet)
+	CHAIN_ID   = test-chain-y3m1e6-AB
+	NETWORK_ID = AB
+	SGX_MODE   = $(sgx_mode)
+else ifeq ($(chain), testnet)
+	CHAIN_ID   = testnet-thaler-crypto-com-chain-42
+	NETWORK_ID = 42
+	SGX_MODE   = HW
+else ifeq ($(chain), mainnet)
+	CHAIN_ID   = thaler-crypto-com-chain-42
+	NETWORK_ID = 42
+	SGX_MODE   = HW
+endif
+
+IMAGE                  = crypto-chain
+IMAGE_RUST_SGX         = rust-sgx
+IMAGE_RUST_CHAIN       = rust-chain
+IMAGE_TENDERMINT       = tendermint/tendermint:v0.32.8
+DOCKER_FILE            = docker/Dockerfile
+DOCKER_FILE_RUST_CHAIN = docker/Dockerfile.rust.chain
+DOCKER_FILE_RUST_SGX   = docker/Dockerfile.rust.sgx
```

Comment:
why are these custom Dockerfiles needed? `cryptocom/chain:latest` from Dockerhub https://hub.docker.com/repository/docker/cryptocom/chain (https://github.com/crypto-com/chain/blob/master/ci-scripts/Dockerfile) should be OK both for development (it has cargo, rustc) as well as quick launching (it has the Intel PSW pre-installed)

Reply:
I'll remove them

Reply:
> integration test fails to build:
> 
> ```
> Successfully built 3f8f9c11031d
> 4886	Successfully tagged integration-tests-chain:latest
> 4887	[2020-01-06|07:00:04] Build Chain Transaction Enclave image
> 4888	[2020-01-06|07:00:04] [Config] SGX_MODE=HW
> 4889	unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat ...../src/chain-tx-enclave/tx-validation/Dockerfile: no such file or directory
> ```

I'll make the integration use this launcher and remove the useless files and scripts.